### PR TITLE
add enable/disable CDN custom domain https

### PR DESCRIPTION
**Pull request by csmengwan**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@csmengwan__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (qinwa). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by markcowl** [comment]
@csmengwan The build is failing because the ENable and Disable commands do not support should process.  If these methods make changes or have side effects, they shoudl support SHouldProcess.

You can implement SHouldProcess correctly by adding SupportsShouldProcess=true to the Cmdlet attribute for the cmdlets, and protecting execution inside a call to ConfirmAction or ShoudlProcess.  Please LMK if this is not clear.

If you think these commands should not support should process, please explain, and add an exception here: https://github.com/Azure/azure-powershell/blob/dev/tools/StaticAnalysis/Exceptions/SignatureIssues.csv

**Comment by cormacpayne** [comment]
@csmengwan ping

File: src/ResourceManager/Cdn/ChangeLog.md [hunk 1]
```diff
@@ -17,7 +17,7 @@
     * Overview of change #1
         - Additional information about change #1
 -->
-## Current Release
+## Current Release (Add disable and enable CDN Custom domain Https)
```
Comment (markcowl) [inline src/ResourceManager/Cdn/ChangeLog.md]: See the examples above.  This should include a bullet for each added cmdlet, i.e.
* Added cmdlets to manage custom domains
  * Disable-AzureRmCdnCustomDomain
  * Remove-AzureRmCdnCustomDomain

File: src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/DisableAzureRmCdnCustomDomainHttps.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        public override void ExecuteCmdlet()
+        {
+            if (ParameterSetName == ObjectParameterSet)
+            {
+                CustomDomainName = CdnCustomDomain.Name;
+                ResourceGroupName = CdnCustomDomain.ResourceGroupName;
+                ProfileName = CdnCustomDomain.ProfileName;
+                EndpointName = CdnCustomDomain.Name;
+            }
+
```
Comment (markcowl) [inline src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/DisableAzureRmCdnCustomDomainHttps.cs]: You need to protect the changes here with a call to ```ConfirmAction``` or ```ShouldProcess```. See: https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#additional-common-methods-to-support-the-change  for details

File: src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/EnableAzureRmCdnCustomDomainHttps.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        {
+            if (ParameterSetName == ObjectParameterSet)
+            {
+                CustomDomainName = CdnCustomDomain.Name;
+                ResourceGroupName = CdnCustomDomain.ResourceGroupName;
+                ProfileName = CdnCustomDomain.ProfileName;
+                EndpointName = CdnCustomDomain.EndpointName;
+            }
+
+            CdnManagementClient.CustomDomains.EnableCustomHttps(ResourceGroupName, ProfileName, EndpointName, CustomDomainName);
```
Comment (markcowl) [inline src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/EnableAzureRmCdnCustomDomainHttps.cs]: Same comment.

File: src/ResourceManager/Cdn/Commands.Cdn/Microsoft.Azure.Commands.Cdn.dll-help.Help.pshproj [hunk 4]
```diff
@@ -1940,5 +1940,179 @@
       <SupportInformation ad="false" rsat="false" ps2="false" ps3="false" ps4="false" ps5="false" wxp="false" wv="false" w7="false" w8="false" w81="false" w10="false" w2k3s="false" w2k3e="false" w2k3d="false" w2k8s="false" w2k8e="false" w2k8d="false" w2k8r2s="false" w2k8r2e="false" w2k8r2d="false" w2k12s="false" w2k12d="false" w2k12r2s="false" w2k12r2d="false" />
```
Comment (markcowl) [inline src/ResourceManager/Cdn/Commands.Cdn/Microsoft.Azure.Commands.Cdn.dll-help.Help.pshproj]: We no longer do help this way, these changes are unnecessary.  Instead, please follow the instructions here: https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md  for creating help in Markdown

**Comment by markcowl** [comment]
@csmengwan closing this PR, it will not make it into this release,a dn there is a moratorium, for 3 weeks after that.

File: src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/EnableAzureRmCdnCustomDomainHttps.cs (hunk 3 above)
Comment (csmengwan) [inline src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/EnableAzureRmCdnCustomDomainHttps.cs]: Our team does not need confirm check for enabling action.

**Pull request closed without merging**